homepage/homepage/content/docs/using-covalues/version-control.mdx

```diff
@@ -0,0 +1,246 @@
+    oninput={handleTitleChange}
+  />
+  
+  {#each project.current?.tasks || [] as task, index}
+    <input 
+      type="text" 
+      value={task.title || ''} 
+      oninput={(e) => handleTaskTitleChange(index, e)}
+    />
+  {/each}
+</form>
+```
+</CodeGroup>
+</ContentByFramework>
+
+### Merging
+
+To merge a branch back to the main version, call `unstable_merge()`:
+
+<CodeGroup>
+```tsx 
+function handleSave() {
+  // Merge the branch changes back to main
+  project.$jazz.unstable_merge();
+  router.navigate("/");
+}
+```
+</CodeGroup>
+
+All the changes made to the branch will be merged into the main CoValue, preserving both author and timestamp.
+
+Calling `unstable_merge()` will merge every change done in the project and the loaded tasks.
+
+The merge is idempotent, so you can merge the same branch multiple times without changing the result.
+
+The merge operation is scoped, so if you call `unstable_merge()` on a task, only the changes to the task will be merged
+
+<CodeGroup>
+```tsx 
+function handleTaskSave(index: number) {
+  const task = project.tasks[index];
+  // Only the changes to the task will be merged
+  task?.$jazz.unstable_merge();
+}
+```
+</CodeGroup>
+
+
+## Conflict Resolution
+
+When conflicts occur (the same field is modified in both the branch and main), Jazz uses a "last writer wins" strategy:
+
+<CodeGroup>
+```tsx
+// Branch modifies priority to "high"
+branch.$jazz.applyDiff({ priority: "high" });
+
+// Meanwhile, main modifies priority to "urgent"
+originalProject.$jazz.applyDiff({ priority: "urgent" });
+
+// Merge the branch
+branch.$jazz.unstable_merge();
+
+// Main's value wins ("urgent") because it was written later
+console.log(originalProject.priority); // "urgent"
+```
+</CodeGroup>
+
+## Private branches
+
+When the owner is not specified, the branch has the same permissions as the main values.
+
+You can also create a private branch by providing a group owner.
+
+<CodeGroup>
+```tsx 
+// Create a private group for the branch
+const privateGroup = Group.create();
+
+const privateBranch = useCoState(Project, projectId, {
+  unstable_branch: { 
+    name: "private-edit", 
+    owner: privateGroup 
+  }
+});
+
+// Only members of privateGroup can see the branch content
+// The sync server cannot read the branch content
+```
+</CodeGroup>
+
+
+You can use private branches both to make the changes to the branches "private" until merged, or to give controlled write access to a group of users.
+
+Only users with both write access to the main branch and read access to the private branch have the rights to merge the branch.
+
+## Branch Identification
+
+You can get the current branch information from the `$jazz` field.
+
+<CodeGroup>
+```ts 
+const branch = await Project.load(projectId, {
+  unstable_branch: { name: "feature-branch" }
+});
+
+// Branch ID is the same as source
+console.log(branch.$jazz.branchName); // "feature-branch"
+console.log(branch.$jazz.isBranch); // true
```

Comment:
I feel like at some point we'll want to make a distinction between the idea of a branch and a "branched value". How do you feel about renaming this method to `isBranched`?